psl/psl-core/src/validate/validation_pipeline/context.rs

```diff
@@ -24,4 +24,9 @@ impl Context<'_> {
     pub(super) fn push_error(&mut self, error: DatamodelError) {
         self.diagnostics.push_error(error);
     }
+
+    /// Pure convenience method. Forwards to Diagnostics::push_warning().
+    pub(super) fn push_warning(&mut self, warning: DatamodelWarning) {
+        self.diagnostics.push_warning(warning);
+    }
 }
```

Comment:
This is also useful for https://github.com/prisma/prisma-private/issues/198 and for https://github.com/prisma/prisma-engines/pull/3435